x-pack/plugins/apm/server/lib/transactions/charts/get_anomaly_data/index.ts

```diff
@@ -39,9 +43,16 @@ export async function getAnomalySeries({
     return;
   }
 
-  // don't fetch anomalies if uiFilters are applied
   if (setup.uiFiltersES.length > 0) {
-    return;
+    // filter out known uiFilters like service.environment & service.name
+    const unknownFilters = setup.uiFiltersES.filter(
+      (uiFilter) =>
+        !uiFilter.term?.[SERVICE_ENVIRONMENT] && !uiFilter.terms?.[SERVICE_NAME]
+    );
+    // don't fetch anomalies if unknown uiFilters are applied
+    if (unknownFilters.length > 0) {
+      return;
+    }
   }
```

Comment:
I think you should pass `query.uiFilters` in from the route, and inspect that instead.